tests/testthat/test-parser.R

```diff
@@ -1,52 +1,54 @@
-  mbox_path_bad <- "/bad/mbox/path"
-  jira_issue_comments_path_bad <- "/bad/jira/path"
-  expect_error(parse_mbox(perceval_path_bad, mbox_path))
-  expect_error(parse_mbox(perceval_path, mbox_path_bad))
-  expect_warning(expect_error(parse_jira(jira_issue_comments_path_bad)))
+test_that("Correct mbox path", {
+  conf_path <-"../../conf/thrift.yml"
+  conf <- yaml::read_yaml(conf_path)
+  mbox_path <- conf[["mailing_list"]][["mbox"]]
+  expect_equal(file.exists(mbox_path), TRUE)
 })
-test_that("Empty file path passed to parse functions fails", {
-  perceval_path <- "../../../tools/bin/perceval"
-  mbox_path <- "../../../../rawdata/mbox/project.mbox"
-  # If a user forgot to set a config parameter, they would have something like
-  # the variable below
-  empty_path <-
-  # What happens if config parameter(s) left blank
-  expect_error(parse_mbox(perceval_path, empty_path))
-  expect_error(parse_mbox(empty_path, perceval_path))
-  expect_error(parse_mbox(empty_path, empty_path))
-  expect_error(parse_jira(empty_path))
+test_that("Correct jira issues comments path", {
+  conf_path <-"../../conf/thrift.yml"
+  conf <- yaml::read_yaml(conf_path)
+  jira_issue_comments_path <- conf[["issue_tracker"]][["jira"]][["issue_comments"]]
+  expect_equal(file.exists(jira_issue_comments_path), TRUE)
 })
-test_that("Absolute filepaths passed to parsers work", {
-  perceval_path_rel <- "../../../tools/bin/perceval"
-  mbox_path_rel <- "../../../../rawdata/mbox/project.mbox"
-  perceval_path_abs <- "/Users/absolute-path/tools/bin/perceval"
-  mbox_path_abs <- "/Users/absolute-path/rawdata/mbox/project.mbox"
-  jira_path_abs <- "/Users/absolute-path/rawdata/issue_tracker/project_issue_comments.json"
-  expect_no_error(parse_mbox(perceval_path_rel, mbox_path_abs))
-  expect_no_error(parse_mbox(perceval_path_abs, mbox_path_rel))
-  expect_no_error(parse_mbox(perceval_path_abs, mbox_path_abs))
-  expect_no_error(parse_jira(jira_path_abs))
+test_that("Incorrect perceval path fails parse_mbox", {
+  conf_path <-"../../conf/thrift.yml"
+  conf <- yaml::read_yaml(conf_path)
+  mbox_path <- conf[["mailing_list"]][["mbox"]]
+  incorrect_perceval_path <- "/incorrect/path/to/perceval"
+  expect_error(parse_mbox(incorrect_perceval_path, mbox_path), "error in running command")
+})
+test_that("Incorrect mbox path fails parse_mbox", {
+  tools_path <- "../../tools.yml"
+  tool <- yaml::read_yaml(tools_path)
+  perceval_path <- tool[["perceval"]]
+  incorrect_mbox_path <- "/incorrect/path/to/mbox"
+  expect_error(parse_mbox(perceval_path, incorrect_mbox_path), "Items of 'old' not found in column names: [data.body]. Consider skip_absent=TRUE.")
+})
+test_that("Incorrect jira issue comments path fails parse_jira", {
+  incorrect_jira_issue_comments_path <- "/incorrect/path/to/jira_issue_comments"
+  expect_error(parse_jira(incorrect_jira_issue_comments_path), "cannot open the connection")
 })
 test_that("Parsers produce tables with correct columns and expected size", {
```

Comment:
I read the review about creating a mini .git file on the fly so that the test remains contained. I'm currently trying to figure that out. I can also delete all references to parse_mbox in this test as well.

Reply:
Hi Nico, 

The function already exists. 

 * http://itm0.shidler.hawaii.edu/kaiaulu/reference/git_create_sample_log.html 

 * http://itm0.shidler.hawaii.edu/kaiaulu/reference/git_delete_sample_log.html

Reply:
I see the function, but I guess I'm just sort of confused. The function creates a sample .git on the fly, but in order to test parse_jira (and no longer parse_mbox), wouldn't I also need the sample repo to have a Jira project key?